Jenkinsfile

```diff
@@ -17,7 +17,7 @@
 // That PR should be landed with out deleting the PR branch.
 // Then a second PR submitted to comment out the @Library line, and when it
 // is landed, both PR branches can be deleted.
-//@Library(value='pipeline-lib@my_branch_name') _
+@Library(value='pipeline-lib@dbohning/daos-15232') _
```

Comment:
Need to backout before merge.

Reply:
Could this work?
```
@Library(value='pipeline-lib@${env.BRANCH_NAME}') _
```

I could test this here, since I need to backout this change and re-push anyway. And if it doesn't work, the commit pragma test will fail. So there's little risk of running too much extra testing if it fails.

Reply:
I am pretty sure we tried that and the result was unfortunately not.  Because it happens so early in the process, I believe it has to be a literal.  No variables or code, etc.

Reply:
I'm not at all literate with Jenkins, but this could work for us?
https://stackoverflow.com/questions/76799131/how-to-test-jenkins-shared-library-with-jenkins-itself

Reply:
Although I imagine there might be a security reason why it doesn't.

Reply:
> I'm not at all literate with Jenkins, but this couldn't work for us? [stackoverflow.com/questions/76799131/how-to-test-jenkins-shared-library-with-jenkins-itself](https://stackoverflow.com/questions/76799131/how-to-test-jenkins-shared-library-with-jenkins-itself)

So yes, disabling implicit loading is one option that I am aware of.  It needs some **thorough** testing on a non-production Jenkins (because it's disabling the global property of disabling implicit library loading) to understand all of it's ramifications though and I don't think our staging Jenkins has been in any usable shape for being able to do this for a long time.

Reply:
Thanks for the explanation. I think probably bigger issues to focus on for now :)

Reply:
Another idea for unit test at least is maybe to put them in a separate groovy script so we don't have to modify Jenkinsfile to update the commit pragma tests